controller/openebs/apiserver.go

```diff
@@ -26,83 +27,118 @@ const (
 	DefaultAPIServerReplicaCount int32 = 1
 )
 
-// MayaAPIServer is a wrapper over k8s deployment structure.
-type MayaAPIServer struct {
-	Object *appsv1.Deployment `json:"object"`
-}
+// updateMayaAPIServer updates the MayaAPIServer manifest as per the reconcile.ObservedOpenEBS values.
+func (p *Planner) updateMayaAPIServer(deploy *unstructured.Unstructured) error {
+	// get the containers of the maya-apiserver and update the desired fields
+	containers, err := unstruct.GetNestedSliceOrError(deploy, "spec", "template", "spec", "containers")
+	if err != nil {
+		return err
+	}
+	// update the env value of containers
+	updateEnv := func(env *unstructured.Unstructured) error {
+		envName, _, err := unstructured.NestedString(env.Object, "spec", "name")
+		if err != nil {
+			return err
+		}
+		if envName == "OPENEBS_IO_INSTALL_DEFAULT_CSTOR_SPARSE_POOL" {
+			unstructured.SetNestedField(env.Object, strconv.FormatBool(
```

Comment:
Errors are not handled? 
SetNestedField returns an error or this is done intentionally?